wgpu/src/lib.rs

```diff
@@ -1936,12 +1931,29 @@ impl Instance {
     /// - On macOS/Metal: will panic if not called on the main thread.
     /// - On web: will panic if the `raw_window_handle` does not properly refer to a
     ///   canvas element.
-    pub unsafe fn create_surface<
+    pub fn create_surface<
+        'window,
+        W: raw_window_handle::HasWindowHandle + raw_window_handle::HasDisplayHandle,
+    >(
+        &self,
+        window: &'window W,
+    ) -> Result<Surface<'window>, CreateSurfaceError> {
+        unsafe { self.create_surface_from_raw(window) }
+    }
+
+    /// See [`create_surface()`](Self::create_surface) for more details.
+    ///
+    /// # Safety
+    ///
+    /// - `raw_window_handle` must be a valid object to create a surface upon.
+    /// - `raw_window_handle` must remain valid until after the returned [`Surface`] is
+    ///   dropped.
+    pub unsafe fn create_surface_from_raw<
         W: raw_window_handle::HasWindowHandle + raw_window_handle::HasDisplayHandle,
```

Comment:
It might actually be more useful to take a [`RawDisplayHandle`](https://docs.rs/raw-window-handle/0.6.0/raw_window_handle/enum.RawDisplayHandle.html) and [`RawWindowHandle`](https://docs.rs/raw-window-handle/0.6.0/raw_window_handle/enum.RawWindowHandle.html) here instead.

Reply:
See https://github.com/rust-windowing/raw-window-handle/pull/139#issuecomment-1703884412.
As far as I interpret this isn't an issue.